src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -172,6 +172,11 @@ private void CancellationCallback()
 
             internal void Start()
             {
+                // Make sure _fullPath doesn't contain a link or alias
+                // since the OS will give back the actual, non link'd or alias'd paths
+                _fullDirectory = Interop.Sys.RealPath(_fullDirectory);
```

Comment:
What if RealPath returns null?  Don't we need to check for that and throw an errno-based error if it does?